test/e2e/generated_clientset.go

```diff
@@ -95,6 +95,9 @@ var _ = Describe("Generated release_1_2 clientset", func() {
 		}
 
 		By("verifying the pod is in kubernetes")
+		time.Sleep(10 * time.Second)
+		selector = labels.SelectorFromSet(labels.Set(map[string]string{"time": value}))
+		options = api.ListOptions{LabelSelector: selector}
```

Comment:
To give you some more overview how it works.
ListOptions has "ResourceVersion" field that you can set. The semantic of this field is as following:
- if this is unset - we will forward list to etcd, without touching watchcache in apiserver
- if this is set - we will serve the result from watchcache in apiserver and guarantee that the returned result is at least as fresh as the given RV

So what was happening before this change was:
- ResourceVersion was set, to a value before "pod creation", to we were serving it from cache without waiting for the pod to be there


So there are two possible fixes:
- set ResourceVersion to 0 (and remove Sleep, because it is then not needed)
- or alternatively: set ResourceVersion in options to the ResourceVersion of create pod, i.e.

pod, err := podClient.Create(pod)
if err != nil {
    Failf("...")
}

By("verifying the pod is in kubernetes")
options.ResourceVersion = pod.ResourceVersion
pods, err := podClient.List(options)


I think the second option is preferred, but I can also live with the first one :)

@lavalamp - FYI